BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/SplashActivity.java

```diff
@@ -1,19 +1,33 @@
 package com.blueprint.foe.beetracker;
 
+import android.content.Context;
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
 
 /**
  * Splash screen shows a bee on a yellow background. This is specified in the SplashTheme.
  */
 public class SplashActivity extends AppCompatActivity {
+    private static final String TAG = SplashActivity.class.toString();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        Intent intent = new Intent(this, HomePageActivity.class);
-        startActivity(intent);
-        finish();
+        SharedPreferences sharedPref = getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+        String token = sharedPref.getString(getString(R.string.preference_login_token), "");
+        boolean isGuest = sharedPref.getBoolean(getString(R.string.preference_logged_in_as_guest), false);
+        Log.d(TAG, token + " " + isGuest);
+        if (!isGuest && token.isEmpty()) {
+            Intent intent = new Intent(this, MainActivity.class);
+            startActivity(intent);
+            finish();
+        } else {
+            Intent intent = new Intent(this, HomePageActivity.class);
+            startActivity(intent);
+            finish();
+        }
```

Comment:
Can consider putting startActivity(intent); finish(); outside if statement.

Or creating a base class where you can put common activity utilities i.e. `startAndFinishActivity(HomePageActivity.class)`